collector/edac_linux.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !noedac
+
+package collector
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"regexp"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	edacSubsystem = "edac"
+)
+
+var (
+	edacMemControllerRE = regexp.MustCompile(`.*devices/system/edac/mc/mc([0-9]*)`)
+	edacMemCsrowRE      = regexp.MustCompile(`.*devices/system/edac/mc/mc[0-9]*/csrow([0-9]*)`)
+)
+
+type edacMCMetric struct {
+	metricName    string
+	metricType    prometheus.ValueType
+	metricHelp    string
+	memController string
+	value         float64
+}
+
+type edacCollector struct {
+	ceCount       *prometheus.Desc
+	ceNoinfoCount *prometheus.Desc
+	ueCount       *prometheus.Desc
+	ueNoinfoCount *prometheus.Desc
+	csrowCeCount  *prometheus.Desc
+	csrowUeCount  *prometheus.Desc
+}
+
+func init() {
+	Factories["edac"] = NewEdacCollector
+}
+
+// Takes a prometheus registry and returns a new Collector exposing
+// edac stats.
+func NewEdacCollector() (Collector, error) {
+	return &edacCollector{
+		ceCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, edacSubsystem, "ce_count"),
```

Comment:
_count is the suffix for Summaries/Histograms. This is probably a counter, so should be _total